app/backend/prepdocs.py

```diff
@@ -167,31 +168,50 @@ def setup_file_processors(
         pdf_parser = LocalPdfParser()
-    else:
+    elif document_intelligence_service is not None:
         pdf_parser = doc_int_parser
+    else:
+        logger.warning("No PDF parser available")
+
+    html_parser: Optional[Parser] = None
     if local_html_parser or document_intelligence_service is None:
         html_parser = LocalHTMLParser()
-    else:
+    elif document_intelligence_service is not None:
         html_parser = doc_int_parser
-    sentence_text_splitter = SentenceTextSplitter(has_image_embeddings=search_images)
-    return {
-        ".pdf": FileProcessor(pdf_parser, sentence_text_splitter),
-        ".html": FileProcessor(html_parser, sentence_text_splitter),
+    else:
+        logger.warning("No HTML parser available")
+
+    # These file formats can always be parsed:
+    file_processors = {
         ".json": FileProcessor(JsonParser(), SimpleTextSplitter()),
-        ".docx": FileProcessor(doc_int_parser, sentence_text_splitter),
-        ".pptx": FileProcessor(doc_int_parser, sentence_text_splitter),
-        ".xlsx": FileProcessor(doc_int_parser, sentence_text_splitter),
-        ".png": FileProcessor(doc_int_parser, sentence_text_splitter),
-        ".jpg": FileProcessor(doc_int_parser, sentence_text_splitter),
-        ".jpeg": FileProcessor(doc_int_parser, sentence_text_splitter),
-        ".tiff": FileProcessor(doc_int_parser, sentence_text_splitter),
-        ".bmp": FileProcessor(doc_int_parser, sentence_text_splitter),
-        ".heic": FileProcessor(doc_int_parser, sentence_text_splitter),
         ".md": FileProcessor(TextParser(), sentence_text_splitter),
         ".txt": FileProcessor(TextParser(), sentence_text_splitter),
     }
+    # These require either a Python package or Document Intelligence
+    if pdf_parser is not None:
+        file_processors.update({".pdf": FileProcessor(pdf_parser, sentence_text_splitter)})
+    if html_parser is not None:
+        file_processors.update({".html": FileProcessor(html_parser, sentence_text_splitter)})
+    # These file formats require Document Intelligence
+    if doc_int_parser is not None:
+        file_processors.update(
+            {
+                ".docx": FileProcessor(doc_int_parser, sentence_text_splitter),
+                ".pptx": FileProcessor(doc_int_parser, sentence_text_splitter),
+                ".xlsx": FileProcessor(doc_int_parser, sentence_text_splitter),
+                ".png": FileProcessor(doc_int_parser, sentence_text_splitter),
+                ".jpg": FileProcessor(doc_int_parser, sentence_text_splitter),
+                ".jpeg": FileProcessor(doc_int_parser, sentence_text_splitter),
+                ".tiff": FileProcessor(doc_int_parser, sentence_text_splitter),
+                ".bmp": FileProcessor(doc_int_parser, sentence_text_splitter),
+                ".heic": FileProcessor(doc_int_parser, sentence_text_splitter),
+            }
+        )
+    return file_processors
```

Comment:
^^ all of that code change was due to issues found via tests/mypy

Reply:
yay mypy